crates/autopilot/src/on_settlement_event_updater.rs

```diff
@@ -111,64 +114,75 @@ impl OnSettlementEventUpdater {
             .try_into()
             .map_err(|err| anyhow!("{}", err))
             .with_context(|| format!("convert nonce {hash:?}"))?;
-        let receipt = self
-            .web3
-            .eth()
-            .transaction_receipt(hash)
-            .await?
-            .with_context(|| format!("no receipt {hash:?}"))?;
-        let gas_used = receipt
-            .gas_used
-            .with_context(|| format!("no gas used {hash:?}"))?;
-        let effective_gas_price = receipt
-            .effective_gas_price
-            .with_context(|| format!("no effective gas price {hash:?}"))?;
 
-        let auction_id = self
-            .db
-            .get_auction_id(tx_from, tx_nonce)
-            .await?
-            .with_context(|| format!("no auction id for tx {hash:?}"))?;
-        let auction_external_prices =
-            self.db
+        let auction_id = self.db.get_auction_id(tx_from, tx_nonce).await?;
+        let mut update = SettlementUpdate {
+            block_number: event.block_number,
+            log_index: event.log_index,
+            tx_from,
+            tx_nonce,
+            ..Default::default()
```

Comment:
Nit: This is slightly error prone when we add a new field because we can forget to add it here and it would be set through default. Since you're just setting one field to None with this, I would do that explicitly.

Reply:
Definitely. Fixed.